register/sha3/multihash_sha3.go

```diff
@@ -0,0 +1,61 @@
+/*
+	This package has no purpose except to perform registration of multihashes.
+
+	It is meant to be used as a side-effecting import, e.g.
+
+		import (
+			_ "github.com/multiformats/go-multihash/register/sha3"
+		)
+
+	This package registers several multihashes for the sha3 family.
+	This also includes some functions known as "shake" and "keccak",
+	since they share much of their implementation and come in the same repos.
+*/
+package sha3
+
+import (
+	"hash"
+
+	"golang.org/x/crypto/sha3"
+
+	"github.com/multiformats/go-multihash"
+)
+
+func init() {
+	multihash.Register(0x14, sha3.New512)
```

Comment:
We really can't use constants this way.

Reply:
Use go-multicodec seem like a plan?

Reply:
Made https://github.com/multiformats/go-multihash/issues/137 to discuss how to solve.